cmd/outline-ss-server/metrics.go

```diff
@@ -161,10 +277,32 @@ func (m *outlineMetrics) SetNumAccessKeys(numKeys int, ports int) {
 	m.ports.Set(float64(ports))
 }
 
-func (m *outlineMetrics) AddOpenTCPConnection(clientInfo ipinfo.IPInfo) {
+func (m *outlineMetrics) AddOpenTCPConnection(addr net.Addr) {
+	clientInfo, err := ipinfo.GetIPInfoFromAddr(m.IPInfoMap, addr)
+	if err != nil {
+		logger.Warningf("Failed client info lookup: %v", err)
+	}
+	logger.Debugf("Got info \"%#v\" for IP %v", clientInfo, addr.String())
```

Comment:
Be careful with this kind of logging. They can have a real impact on memory allocations. I'd rather remove it or protect with a if block.